text/0000-ember-new-lang.md

```diff
@@ -0,0 +1,239 @@
+
+```bash
+ember help new
+ember new <app-name> <options...>
+  ...
+  --lang (String) (Default: "") Sets the base human language of the application via index.html
+    alias: -l <value>
+```
+
+### Misusage and Error Handling
+
+
+Broadly, incorrect usage of the `--lang` flag covers three use-case categories:
+
+- The flag has been specified with a programming language as the argument
+- The flag has been specified with no argument
+- The flag has been specified with an invalid language code
+
+This RFC proposes that these cases cause the build process to halt with a revelant error and help message as opposed to simply reverting to the default value and reporting a message. **The rationale for the recommendation to halt instead of just report is that in these all of these cases, the user has explicitly typed `--lang` into their CLI tool. This is an unambiguous declaration of intent by the user to use the `--lang` flag correctly.**
+
+
+
+#### Invalid Language Codes
+
+Language codes are verified against [is-language-code](https://www.npmjs.com/package/is-language-code). (see [examples of valid ISO country codes](https://en.wikipedia.org/wiki/List_of_ISO_3166_country_codes))
+
+If an invalid language code is given such as `--lang en-UK` the indended output should be a shell error that will halt the build.
+
+```bash
+ember new my-app --lang en-UK
+Unrecognized language subtag, "uk".
+```
+
+#### No argument passed
+Additionally, if the `--lang` flag is used but no argument is defined, e.g., `ember new my-app --lang`, the build will also be halted. An error message and usage info should be shown in these cases:
+
+```bash
+# Detect trailing option (declared)
+ember new my-app --lang --skip-git
+
+An error with the `--lang` flag returned the following message:
+  Detected lang specification starting with command flag `-`.
+  Is `--skip-git` meant to be an ember-cli command option?
+  This issue is likely caused by using the `--lang` flag without a specification.
+Information about using the `--lang` flag:
+  The `--lang` flag sets the base human language of the app in index.html
+  If used, the lang option must specfify a valid language code.
+  For default behavior, remove the flag.
+  See `ember <command> help` for more information.
+```
+
+```bash
+# Detect trailing option (undeclared)
+ember new my-app --lang
+
+An error with the `--lang` flag returned the following message:
+  Detected lang specification starting with command flag `-`.
+  Is `--disable-analytics` meant to be an ember-cli command option?
+  This issue is likely caused by using the `--lang` flag without a specification.
+Information about using the `--lang` flag:
+  The `--lang` flag sets the base human language of the app in index.html
+  If used, the lang option must specfify a valid language code.
+  For default behavior, remove the flag.
+  See `ember <command> help` for more information.
+```
+
+#### Common Misunderstandings: Programming Languages
```

Comment:
IMO this is just a special case of the "Invalid Language Codes". In implementation speak:

```js
let lang;

try {
  // validateLangFlag checks against the ISO 3166 list
  lang = validateLangFlag(langFlag);
} catch(e) {
  if (isProgrammingLanguage(langFlag)) {
    throw new Error("specific error message about how this flag is not doing what you think");
  } else {
    throw new Error("generic error message about how this is not a valid language");
  }
}
```

In the unlikely case that they overlap (that it is both a valid language and a programming language) I think it's okay to print a warning, but probably shouldn't fail the build and disallow what is technical a valid input.